pin-project 0.4.11 breaks hyper

Hello, hyper uses `#[project]` on an expression https://github.com/hyperium/hyper/blob/v0.13.5/src/server/conn.rs#L926 and https://github.com/hyperium/hyper/blob/v0.13.5/src/server/shutdown.rs#L65 which now causes a compilation error with pin-project 0.4.11.

Here is one of your test cases adapted to demonstrate the issue, this code works fine with 0.4.10 but not 0.4.11:

```rust
#[project]
fn foo() {
    #[pin_project]
    enum Enum<A> {
        Variant(#[pin] A),
    }

    let mut x = Enum::Variant(1);
    let mut x = Pin::new(&mut x).project();

    let asdf = {
        #[project]
        match x {
            Enum::Variant(x) => {}
        }
    };

    Some(asdf);
}
```

Error:

```
error[E0658]: attributes on expressions are experimental
  --> src/main.rs:16:9
   |
16 |         #[project]
   |         ^^^^^^^^^^
   |
   = note: see issue #15701 <https://github.com/rust-lang/rust/issues/15701> for more information
   = help: add `#![feature(stmt_expr_attributes)]` to the crate attributes to enable
```

--- a/pin-project-internal/src/project.rs
+++ b/pin-project-internal/src/project.rs
@@ -208,10 +208,7 @@ fn replace_item_fn(item: &mut ItemFn, mutability: Mutability) -> Result<()> {
 
         fn visit_stmt(&mut self, node: &mut Stmt) -> Result<()> {
             match node {
-                Stmt::Expr(expr) | Stmt::Semi(expr, _) => {
-                    visit_mut::visit_expr_mut(self, expr);
-                    self.visit_expr(expr)
-                }
+                Stmt::Expr(expr) | Stmt::Semi(expr, _) => self.visit_expr(expr),
                 Stmt::Local(local) => {
                     visit_mut::visit_local_mut(self, local);
                     if let Some(attr) = local.attrs.find_remove(self.name())? {
@@ -226,6 +223,7 @@ fn replace_item_fn(item: &mut ItemFn, mutability: Mutability) -> Result<()> {
         }
 
         fn visit_expr(&mut self, node: &mut Expr) -> Result<()> {
+            visit_mut::visit_expr_mut(self, node);
             let attr = match node {
                 Expr::Match(expr) => expr.attrs.find_remove(self.name())?,
                 Expr::If(expr_if) => {

--- a/tests/project.rs
+++ b/tests/project.rs
@@ -204,3 +204,33 @@ fn non_stmt_expr_match() {
         },
     );
 }
+
+// https://github.com/taiki-e/pin-project/issues/206
+#[test]
+#[project]
+fn issue_206() {
+    #[pin_project]
+    enum Enum<A> {
+        Variant(#[pin] A),
+    }
+
+    let mut x = Enum::Variant(1);
+    let x = Pin::new(&mut x).project();
+
+    Some({
+        #[project]
+        match &x {
+            Enum::Variant(_) => {}
+        }
+    });
+
+    loop {
+        let _ = {
+            #[project]
+            match &x {
+                Enum::Variant(_) => {}
+            }
+        };
+        break;
+    }
+}

